odoo/addons/base/models/ir_asset.py

```diff
@@ -392,6 +392,10 @@ def _process_command(self, command):
             target = None
         return directive, target, path_def
 
+    def _load_records_write(self, values):
+        values.pop('active', None)
+        super()._load_records_write(values)
```

Comment:
@odoo/rd-security @odoo/rd-framework-py What do you think of this? The discussion started here: https://github.com/odoo/upgrade/pull/3829#issuecomment-1297363594 As suggested, maybe we should also change the way this is done for views (not relying on the use of `<template>`, but doing it like here) ? https://github.com/odoo/odoo/blob/f1c70d4cc943ac4eb81a85a9dc005de34cd2060a/openerp/tools/convert.py#L801

@bso-odoo Could be nice to mention more things in the commit message (the fact that it was the case for "assets views", the link to the code doing that, the link of the original commit I mentioned which added the active fields, etc ;))

Reply:
> What do you think of this?

:shrug: it's not great but I guess the updating the assets via the XML files is not really a use-case?

> As suggested, maybe we should also change the way this is done for views (not relying on the use of `<template>`, but doing it like here) ?

Is there a use-case for that? Because it seems a lot less innocuous, as I could well see the ability to update view activation being important for module evolution / migration, and if a user wants to craft their updates around this, they can use custom import `context` and `<record>` tags.

Reply:
> Maybe not (although could make sense as explained above), but in that case, should we rely on the use of `<template>`?

We've been doing that for 8 years so... maybe?

>  I don't see a good reason to not have the same behavior if that qweb view is defined using `<record>`?

And I don't see any reason to have it. If you're using raw `record` you can already achieve that using `default_` keys in the context.

Reply:
> We've been doing that for 8 years so... maybe?

And hopefully we improved the codebase for 8 years :wink: 

> If you're using raw record you can already achieve that using default_ keys in the context.

I don't know what you mean, can you explain? :grimacing: 

> And I don't see any reason to have it.

Probably not critical but I have 3 :smile: 
- Consistency: I really don't understand why would `<template>` be different from `<record>`. I am pretty convinced that most users only see it as a shortcut for `<record>`, not something with additional behaviors. I would be surprised if the error was not there in custom modules. And as we plan to develop the documentation about creating custom website modules/themes to attract developers and improve our PS tech designer team tools, the more consistent the better.
- Readability: I don't know how much time he took me to find that piece of code but way too long (and actually, I found it 2 years ago and it took me about the same time to find it again last week).
- Robustness: Kinda related to the previous point but having this behavior anchored to the views python file via a dedicated python method would prevent this behavior to be lost / not adapted in a refactoring. Maybe if it was there, the behavior would not have been lost when creating the `ir.asset` model.

Again nothing critical (I would actually not "fix" that in stable, but in master).

Reply:
> > If you're using raw record you can already achieve that using default_ keys in the context.
> 
> I don't know what you mean, can you explain? grimacing

`default_` keys in contexts only apply to create calls. The record (and function) elements support `uid` and `context` attributes to customise their environment.

> * Consistency: I really don't understand why would `<template>` be different from `<record>`. I am pretty convinced that most users only see it as a shortcut for `<record>`

Which it is.

> not something with additional behaviors.

The two are not exclusive? `<template>` is strictly less flexible than `<record>`. That means if there are specific use cases were want to support in `<template>` they may need to... be specifically supported there. That doesn't mean `<record>` needs them.

Meanwhile if you move this to the model you're now fundamentally altering the way it works globally.

Reply:
I hear you, but there is no way lambda developers guess that and even with a good tutorial, I still don't think it's good in practice: having to explain that you can use `<record>` to create records but for qweb views, there is high probability that you should use `<template>` to not break future updates and that even in the case you would want to use `<record>` for your specific usecase, it would not work as you would have to write an upgrade script to make it work... meh.

> Meanwhile if you move this to the model you're now fundamentally altering the way it works globally.

**during updates** (which is my point: updating the active field of a qweb view during updates does not work ever for frontend related layouts without upgrade scripts, so I would always disabling it by default during update). But ok, I guess we should not support qweb views are only used for frontend layouts.

cc @bso-odoo If you can at least advertise that behavior next to the `active` field definition of the ir.ui.view model ("is not updated during updates if `<template>` is used instead of `<record>`, see `_tag_template`" something like that), that would be nice :+1:). (An other PR in 14.0 mentioning that discussion).

Reply:
@rdeodoo For this current PR here, with `ir.asset`, what do you think? Should we use `_load_records_write` as in the current state, or should we try to use `default_` keys in context definition of all `<record>` to indicate `active` should only be set to `False` during creation and not update. In the end, if it makes sense to keep the standard behavior of `<record>` for views, I think it makes sense to keep it for `ir.asset` too. Or we could also create `<asset>`, to behave just like `<template>`. Probably what I would vote for: that way it's consistent.

Reply:
In order of preference:
1. Have `<template/>` and `<record/>` behaves the same (the active field bypass also occurs with `<record/>`), it's very confusing that the shortcut version is by passing this field, it is just bug prone.
And use `_load_records_write` for both I guess or something like that
2. If framework is against is and wants to keep the inconsistency between `<record/>` and `<template/>` , indeed an `<asset>` tag to mimick that `active` field bypass with the shortcut tag would be the second best choice to keep to consistency
3. current state with only `_load_records_write`
4. `default_` stuff

Reply:
@bso-odoo You can adapt with second choice.
If the framework accepts to align `<template/>` and `<record/>` later we will easily be able to do the same, + it's always nice to have a shortcut for this

Reply:
@rdeodoo I introduced the `<asset>` tag, and adapted all (~100) existing `<record ... model="ir.asset">` records.
I will have to make sure I scan for those again at each forward-port.
Note that there was one such record in enterprise => odoo/enterprise#35153.

The rule is the same as for the `<template>` tag:
- if `active` is defined as an attribute on the `<asset>` node, apply it only for new records (otherwise, do not talk about it in the created record)

Reply:
@bso-odoo Don't you need to do it for themes too alongside making sure it works for theme.ir.asset as well? :wink: Or is this working in themes already because they are theme.ir.asset? (I guess the current theme.ir.asset are compatibility stuff so not as important (although still need to be fixed the same way if the bug is the same) but custom themes would create snippets using that)

Reply:
Themes indeed need to be adapted and we should also convert those to `<assets/>` and adapt the `_tag_asset` new method to handle it like the `_tag_template` does for templates -> Map `<assets/>` to `theme.ir.asset` if declared in a theme module
```py
        if self.module.startswith('theme_'):
            model = 'theme.ir.asset'
        else:
            model = 'ir.asset'
```

Reply:
@qsm-odoo @rdeodoo Now handling `theme.ir.asset` too, additional PR: https://github.com/odoo/design-themes/pull/629.
Also note that I added `directive` as an attribute of `bundle`.